test/http_sender.js

```diff
@@ -283,4 +284,44 @@ describe('http sender', () => {
       done();
     });
   });
+
+  it('should avoid OOM on failed buffer conversion', done => {
```

Comment:
On closer inspection, I don't think this test adds much in addition to the assert in L232, and it also runs for a long time (5s), so let's remove this.

Reply:
That's what I thought at first. So I both add the assert in L232 and put this unit test as a comment which may be helpful to understand the issue.